apps/ogr2ogr_lib.cpp

```diff
@@ -4247,41 +4320,107 @@ SetupTargetLayer::Setup(OGRLayer *poSrcLayer, const char *pszNewLayerName,
             papszLCOTemp =
                 CSLSetNameValue(papszLCOTemp, "GEOMETRY_NULLABLE", "NO");
             CPLDebug("GDALVectorTranslate", "Using GEOMETRY_NULLABLE=NO");
         }
 
-        // Use source geometry field name as much as possible
-        if (eGType != wkbNone && pszDestCreationOptions &&
-            strstr(pszDestCreationOptions, "GEOMETRY_NAME") != nullptr &&
-            CSLFetchNameValue(m_papszLCO, "GEOMETRY_NAME") == nullptr)
+        if (psOptions->dfXYRes != OGRGeomCoordinatePrecision::UNKNOWN)
         {
-            int iSrcGeomField = -1;
-            if (anRequestedGeomFields.empty() &&
-                (nSrcGeomFieldCount == 1 ||
-                 (!m_poDstDS->TestCapability(
-                      ODsCCreateGeomFieldAfterCreateLayer) &&
-                  nSrcGeomFieldCount > 1)))
+            if (m_poDstDS->GetDriver()->GetMetadataItem(
+                    GDAL_DCAP_HONOR_GEOM_COORDINATE_PRECISION) == nullptr &&
+                !OGRGeometryFactory::haveGEOS())
             {
-                iSrcGeomField = 0;
+                CPLError(CE_Warning, CPLE_AppDefined,
+                         "-xyRes specified, but driver does not expose the "
+                         "DCAP_HONOR_GEOM_COORDINATE_PRECISION capability, "
+                         "and this build has no GEOS support");
             }
-            else if (anRequestedGeomFields.size() == 1)
+
+            oCoordPrec.dfXYResolution = psOptions->dfXYRes;
+            if (!psOptions->osXYResUnit.empty())
             {
-                iSrcGeomField = anRequestedGeomFields[0];
+                if (!poOutputSRS)
+                {
+                    CSLDestroy(papszLCOTemp);
+                    CPLError(CE_Failure, CPLE_AppDefined,
+                             "Unit suffix for -xyRes cannot be used with a "
+                             "unknown destination SRS");
+                    return nullptr;
+                }
+
+                if (psOptions->osXYResUnit == "mm")
+                {
+                    oCoordPrec.dfXYResolution *= 1e-3;
+                }
+                else if (psOptions->osXYResUnit == "deg")
+                {
+                    double dfFactorDegToMetre =
+                        poOutputSRS->GetSemiMajor(nullptr) * M_PI / 180;
+                    oCoordPrec.dfXYResolution *= dfFactorDegToMetre;
+                }
+                else
+                {
+                    // Checked at argument parsing time
+                    CPLAssert(psOptions->osXYResUnit == "m");
+                }
+
+                OGRGeomCoordinatePrecision tmp;
+                tmp.SetFromMeter(poOutputSRS, oCoordPrec.dfXYResolution, 0, 0);
+                oCoordPrec.dfXYResolution = tmp.dfXYResolution;
             }
+        }
 
-            if (iSrcGeomField >= 0)
+        if (psOptions->dfZRes != OGRGeomCoordinatePrecision::UNKNOWN)
+        {
+            if (m_poDstDS->GetDriver()->GetMetadataItem(
```

Comment:
I believe we mentioned to use ``!x`` only for variables, and not for more complex statements, didn't we? Anyway I'm not used to that convention, and most of the codebase used "x == nullptr" convention. There are clang-tidy rules actually preferring that "x == nullptr" convention. I got actually bitten by that when contributing to https://github.com/p-ranav/argparse/pull/333 since their clang-tidy config enables https://clang.llvm.org/extra/clang-tidy/checks/readability/implicit-bool-conversion.html

Reply:
I prefer the shortest (implicit) version, but at the end of the day we should just try to be consistent.

Reply:
I'm not against having some policy regarding that, but my "conditions" would be that we have hopefully a pre-commit way of enforcing it, or some easy automation, otherwise it won't be followed. That might be harder to do than just styling though as the clang-tidy checks require a form of code compilation

Reply:
I'm also happy with stopping to bother about that if we can't find a clang-tidy way to enforce it, it's not so important.